content/articles/python-app-that-extracts-images-scan-text-transcribe-save-to-txt-file/index.md

```diff
@@ -0,0 +1,224 @@
+
+To install PyMuPDF, run the following command:
+
+```bash
+pip install PyMuPDF
+```
+
+- Pillow
+
+[Pillow](https://pypi.org/project/Pillow/) provides image interpreter and processing capabilities for Python. To install pillow, run the following command:
+
+```bash
+pip install pillow
+```
+
+- Opencv-python
+
+[Opencv-python](https://pypi.org/project/opencv-python/) is used to read images and video, manipulate media files with image transformations, draw shapes and put text on those files. We will use Opencv to text recognition inside media files (images). This will help to detect and find text information in an image and recognize them using inbuilt methods.
+
+To install opencv-python, run the following command:
+
+```bash
+pip install opencv-python
+```
+
+### Create a Python tesseract script
+
+Let's dive in and write a script to execute this logic using python. Create a project folder and add a new `main.py` file inside that folder. Below is we will create this application:
+
+First, we need to import these library dependencies that we installed in the above `pip` commands. Add the following imports inside the `main.py` file:
+
+```py
+import os  # from native modules
+import fitz  # from PyMuPDF
+import pytesseract  # from pytesseract
+import cv2  # from Opencv
+import io  # from native modules
+from PIL import Image, ImageFile  # from Pillow
+from colorama import Fore  # from native modules
+import platform  # from native modules
+```
+
+Then allow this application to process the image files:
+
+```py
+ImageFile.LOAD_TRUNCATED_IMAGES = True
+```
+
+Once the application access a PDF, its content will be extracted in the form of images. These images will then be processed to extract the text. In this case, we need to create some global variables that help create and save these images to the project path. We also need the path to save the extracted text into a `.txt` file. Go ahead and add these global variables:
+
+```py
+# Global variables
+strPDF, textScanned,  textScanned, inputTeEx, dirName = "", "", "", "", [
+    "images", "output_txt"]
+```
+
+This will create a directory `images` where the PDF extracted images will be saved. An `output_txt` director will also be created to save the scanned text information.
+
+```py
+def gInUs():
+    # Global var
+    global strPDF
+    global inputTeEx
+
+    if(platform.system() == "Windows"):
+        print(Fore.YELLOW +
+              "[.] Insert path to your tesseract.exe" + Fore.RESET)
+        inputTeEx = input()
+
+    # Print input
+    print(Fore.GREEN + "[!] Insert path to PDF file:" + Fore.RESET)
+    inputUser = input()
+
+    # Print an alert if input is not valid, if not, call to fun reDoc
+    if(inputUser == "" or len(inputUser.split("\\")) == 1):
+        print(Fore.RED + "[X] Please put a valid PATH to a file" + Fore.RESET)
+    else:
+        extIm(inputUser)
+```
```

Comment:
Code explanation - state objectives of this code.

Put them as bulleted points